astropy/io/fits/fitstime.py

```diff
@@ -175,22 +214,71 @@ def _verify_column_info(column_info, global_info):
             column_info['scale'] = global_info['scale']
             column_info['format'] = global_info.get('format', None)
 
-        # Non-linear coordinate types have "4-3" form and are not time coordinates
-        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+        else:
+            # Coordinate type is either an unrecognized local time scale
+            # or a linear coordinate type
             return False
 
-        # Coordinate type is either an unrecognized local time scale
-        # or a linear coordinate type
-        return False
+    elif (unit is not None and unit in FITS_TIME_UNIT) or location is not None:
```

Comment:
OK, after some looking through the documentation, I understand now that `TCUNI` is specifically for coordinate units, so if that is a time unit, it is fine to assume the column is a time.